Source/Core/UICommon/UICommon.cpp

```diff
@@ -316,7 +306,13 @@ void SetUserDirectory(std::string custom_path)
     // 4. Default
     //    -> Use XDG basedir, see
     //    http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
-    user_path = home_path + "." DOLPHIN_DATA_DIR DIR_SEP;
+    //
+    // On Android or macOS:
+    // 1. GetExeDirectory()/portable.txt exists
+    //    -> Use GetExeDirectory()/User
```

Comment:
We should probably not check for portable.txt on Android. It's kind of a nonsense operation.

Reply:
What about ``DOLPHIN_EMU_USERPATH``?

Reply:
I don't think that has any practical use on Android. But `getenv("DOLPHIN_EMU_USERPATH")` should at least be faster than hitting the file system.

Reply:
Ah, I forgot. On Android we always set `custom_path` when calling this function, so it actually doesn't matter what we do.

Reply:
Sounds good. Since this part of the coded doesn't matter for Android, I reverted back to my previous commit since trying to split ``ifdef ANDROID`` made things a bit messy.